src/shop.ts

```diff
@@ -58,6 +62,39 @@ export class Shop {
       timestamp: new Date().getTime(),
     }
 
-    return await this.storage.store(order)
+    return await this.storage.store(order, this.configuration.publicKey)
+  }
+
+  // back-end
+
+  async list(privateKey: JsonWebKey): Promise<Order[]> {
+    return this.storage.list(privateKey)
+  }
+
+  // setup
+
+  static async generateCrypto(): Promise<ShopCrypto> {
+    const crypto = new WebCrypto()
+    const pair = await crypto.generateKeyPair()
+
+    const privateKey = await crypto.exportKey(pair.privateKey)
+    const publicKey = await crypto.exportKey(pair.publicKey)
+    const id = await crypto.hash(publicKey.n)
+
+    return { privateKey, publicKey, id }
+  }
+
+  static generateConfiguration(publicKey: JsonWebKey, id: string): string {
+    const config: ShopConfiguration = {
+      name:
+        'A name for your shop - customers will see this during the payment - short and easy to recognize',
+      address: 'Your Nimiq address to receive payments on',
+      logo:
+        'URL of a 128x128px PNG file showing the logo of your shop on transparent background (optional)',
+      live: false,
+      id,
+      publicKey,
+    }
```

Comment:
What's the goal to have a `ShopConfiguration` with invalid `name`, `logo` and `address`. Why not adding a comment or better a documentation what ``name`, `logo` and `address` are needed for?

Reply:
Yes, fully agree. This is a quick solution to get a minimal setup going where the shop owner edits the JSON in a textarea, the values are then hints of what to do - minimal instructions are there, too. V2 should have a proper form and everything. Wdyt, OK for V1?

Reply:
Sure, absolutely fine for a MVP `V1`